src/lib/hooks/useCurrency.ts

```diff
@@ -46,6 +48,11 @@ export function useTokenFromActiveNetwork(tokenAddress: string | undefined): Tok
   const symbol = useSingleCallResult(tokenContract, 'symbol', undefined, NEVER_RELOAD)
   const symbolBytes32 = useSingleCallResult(tokenContractBytes32, 'symbol', undefined, NEVER_RELOAD)
   const decimals = useSingleCallResult(tokenContract, 'decimals', undefined, NEVER_RELOAD)
+  sendAnalyticsEvent(InterfaceEventName.WALLET_PROVIDER_USED, {
```

Comment:
so this one's a bit tricky i think, basically this event doesn't really measure how many times the `provider` is actually used. so `useSingleCallResult` uses rtk-query under the hood and the results are cached per block number. and here the `NEVER_RELOAD` flag means that the results are cached for the rest of the session. so it doesn't necessarily mean that every time this hook runs, the provider is used

im not super sure how to fix tbh, maybe an alternative would be to toss this in a `useEffect` where we estimate that the provider is only used every time a set of variable changes

Reply:
I've moved the call up 1 level and wrapped it in a `useEffect` based on if the token retrieved changes and have the event only firing once when querying for a token
ex:  `0xf9c5ce0c5fa29b487b7329bb92d8e473c386f82b`
<img width="1727" alt="Screenshot 2023-08-23 at 2 35 35 PM" src="https://github.com/Uniswap/interface/assets/11512321/e592fe5c-9f06-409f-96c0-27374e2fe396">

There's still the chance if the user continually queries the same token in the same session that the number might be slightly inflated but I think a marginal overestimate is acceptable